fav/fav.go

```diff
@@ -0,0 +1,182 @@
+package fav
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+	"github.com/meateam/api-gateway/factory"
+	"github.com/meateam/api-gateway/oauth"
+	"github.com/meateam/api-gateway/user"
+	"google.golang.org/grpc/status"
+	"github.com/sirupsen/logrus"
+	gwruntime "github.com/grpc-ecosystem/grpc-gateway/runtime"
+	loggermiddleware "github.com/meateam/api-gateway/logger"
+	fvpb "github.com/meateam/fav-service/proto"
+	fpb "github.com/meateam/file-service/proto/file"
+	grpcPoolTypes "github.com/meateam/grpc-go-conn-pool/grpc/types"
+
+)
+
+const (
+	// ParamFileID is the name of the file id param in URL.
+	ParamFileID = "id"
+)
+
+// Fav is a struct that describes a user's request to a favorite file.
+type Fav struct {
+	UserID string `json:"userID,omitempty"`
+	FileID string `json:"fileID,omitempty"`
+}
+
+// Router is a structure that handles favorite requests.
+type Router struct {
+	favClient       factory.FavClientFactory
+	fileClient 		factory.FileClientFactory
+	oAuthMiddleware *oauth.Middleware
+	logger          *logrus.Logger
+}
+
+// NewRouter creates a new Router, and initializes clients of the quota Service
```

Comment:
```suggestion
// NewRouter creates a new Router, and initializes clients of the fac-service
```
When you copy-paste... :eyes:

Reply:
Done